src/instructlab/model/serve_backend.py

```diff
@@ -78,14 +116,25 @@ def serve_backend(
             log_file=log_file,
         )
     elif backend == backends.VLLM:
+        # Third Party
+        import torch
+
         # First Party
         from instructlab.cli.model.serve import warn_for_unsupported_backend_param
         from instructlab.utils import contains_argument
 
         warn_for_unsupported_backend_param(ctx)
 
         ctx.obj.config.serve.vllm.vllm_args = ctx.obj.config.serve.vllm.vllm_args or []
+
+        available_gpus = torch.cuda.device_count()
         if gpus:
+            gpus_valid = check_gpu_count(gpus, available_gpus)
```

Comment:
I would remove the call to `check_gpu_count`, it seems like this is something you can just handle on the line below.

Reply:
use function, seems just easy to use for the test...

Reply:
Removed.